File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/storage/TripPinsStorage.java
package com.worldventures.dreamtrips.modules.trips.storage;

import android.support.annotation.Nullable;

import com.worldventures.dreamtrips.core.janet.cache.CacheBundle;
import com.worldventures.dreamtrips.core.janet.cache.CachedAction;
import com.worldventures.dreamtrips.core.janet.cache.storage.ActionStorage;
import com.worldventures.dreamtrips.core.repository.SnappyRepository;
import com.worldventures.dreamtrips.modules.trips.command.GetTripsLocationsCommand;
import com.worldventures.dreamtrips.modules.trips.model.Pin;

import java.util.List;

public class TripPinsStorage implements ActionStorage<List<Pin>> {

   private final SnappyRepository snappyRepository;

   public TripPinsStorage(SnappyRepository snappyRepository) {
      this.snappyRepository = snappyRepository;
   }

   @Override
   public Class<? extends CachedAction> getActionClass() {
      return GetTripsLocationsCommand.class;
   }

   @Override
   public void save(@Nullable CacheBundle params, List<Pin> data) {
      snappyRepository.savePins(data);
   }

   @Override
   public List<Pin> get(@Nullable CacheBundle action) {
      return snappyRepository.getPins();
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/service/storage/TripPinsStorage.kt
package com.worldventures.dreamtrips.modules.trips.service.storage

import com.worldventures.dreamtrips.core.repository.SnappyRepository
import com.worldventures.dreamtrips.modules.trips.model.map.Pin
import com.worldventures.dreamtrips.modules.trips.service.command.GetTripsLocationsCommand
import com.worldventures.janet.cache.CacheBundle
import com.worldventures.janet.cache.storage.ActionStorage

open class TripPinsStorage(private val snappyRepository: SnappyRepository) : ActionStorage<List<Pin>> {

   override fun getActionClass() = GetTripsLocationsCommand::class.java

   override fun save(params: CacheBundle?, data: List<Pin>) = snappyRepository.savePins(data)

   override fun get(action: CacheBundle?) = snappyRepository.pins
}
